src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,177 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  Button, Form, Hyperlink, ModalLayer, Spinner, TransitionReplace,
+} from '@edx/paragon';
+import classNames from 'classnames';
+import { useFormik } from 'formik';
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import * as Yup from 'yup';
+import FormSwitchGroup from '../../generic/FormSwitchGroup';
+import { useModel } from '../../generic/model-store';
+import StatusBadge from '../../generic/status-badge/StatusBadge';
+import { LOADED, LOADING } from '../data/slice';
+import { enableDisableApp } from '../data/thunks';
+import AppConfigFormDivider from '../discussions/app-config-form/apps/shared/AppConfigFormDivider';
+import { PagesAndResourcesContext } from '../PagesAndResourcesProvider';
+import messages from './messages';
+
+/**
+ *
+ * @param {string} appId
+ * @param {{}} initialValues
+ * @param {{}} validationSchema
+ * @param {function}onSubmit
+ */
+export function useAppSettingsForm({
+  appId,
+  initialValues = {},
+  validationSchema = {},
+  onSubmit = null,
+}) {
+  const { courseId } = useContext(PagesAndResourcesContext);
+  const appInfo = useModel('courseApps', appId);
+  const dispatch = useDispatch();
+  const handleSubmit = (values) => {
+    // If the app's enabled/disabled status has changed, set that first.
+    if (appInfo.enabled !== values.enabled) {
+      dispatch(enableDisableApp(courseId, appInfo.id, values.enabled, true));
+    }
+    // Call the submit handler for the settings component to save its settings
+    if (onSubmit) {
+      onSubmit();
+    }
+  };
+  return useFormik({
+    initialValues: {
+      enabled: !!appInfo?.enabled,
+      ...initialValues,
+    },
+    validationSchema: Yup.object()
+      .shape({
+        enabled: Yup.boolean(),
+        ...validationSchema,
+      }),
+    onSubmit: handleSubmit,
+  });
```

Comment:
useFormik limits us sometimes we are restricted in many features e.g 
Field, FastField, ErrorMessage, connect(), and FieldArray will NOT work with useFormik() as they all require React Context you can learn more [here](https://formik.org/docs/api/useFormik) 
 Considering future perspective its best to use `<Formik />` component instead which is more powerful. 
 
 As you can see in figma when you will need to work on team settings you will need to work with `<FieldArray>` then useFormik will not work. for help you can take help from [this](https://github.com/edx/frontend-app-course-authoring/blob/master/src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/DiscussionTopics.jsx) already implemented example

Reply:
any thought on this one?

Reply:
I will investigate this and update the PR accordingly.

Reply:
I've updated the code to use Formik as you suggested.